src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfSimpleTypesTests.cs

```diff
@@ -166,5 +167,51 @@ public void TestUdfWithReturnAsTimestampType()
                 }
             }
         }
+
+        /// <summary>
+        /// UDFs defined in separate threads to validate ThreadStatic bug fix.
+        /// </summary>
+        [Fact]
+        public void TestUdfWithMultipleThreads()
+        {
+            try
+            {
+                static void Method1()
+                {
+                    try
+                    {
+                        Func<Column, Column> udf1 = Udf<string, string>(
+                        str => $"{str} - test udf1");
+                    }
+                    catch (Exception e)
+                    {
+                        throw new Exception(e.ToString());
+                    }
+                }
+
+                static void Method2()
+                {
+                    try
+                    {
+                        Func<Column, Column> udf2 = Udf<string, string>(
+                        str => $"{str} - test udf2");
+                    }
+                    catch (Exception e)
+                    {
+                        throw new Exception(e.ToString());
+                    }
+                }
+
+                Thread t1 = new Thread(Method1);
+                Thread t2 = new Thread(Method2);
+                t1.Start();
+                t1.Join();
+                t2.Start();
```

Comment:
make sure this code path is not hit (meaning it already threw).

Reply:
Yes this code path is not being hit while debugging, I checked, it throws the exception at the end of Method1, but while running, it does not catch it in the outer try/catch block but just aborts. Trying with one thread